src/librustc/mir/repr.rs

```diff
@@ -850,6 +851,16 @@ pub enum BinOp {
     Gt,
 }
 
+impl BinOp {
+    pub fn is_checkable(self) -> bool {
+        use self::BinOp::*;
+        match self {
+            Add | Sub | Mul | Shl | Shr => true,
```

Comment:
The `_` could be expanded to all the various options to avoid divergence concerns.